lib/web/desktop.go

```diff
@@ -539,18 +542,108 @@ func (c *connector) tryConnect(ctx context.Context, clusterName, desktopServiceI
 	return conn, ver, trace.Wrap(err)
 }
 
+// desktopPinger measures latency between proxy and the desktop by sending tdp.Ping messages
+// Windows Desktop Service and measuring the time it takes to receive message with the same UUID back.
+type desktopPinger struct {
+	wds net.Conn
+	ch  <-chan tdp.Ping
+}
+
+func (d desktopPinger) Ping(ctx context.Context) error {
+	ping := tdp.Ping{
+		UUID: uuid.New(),
+	}
+	buf, err := ping.Encode()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	_, err = d.wds.Write(buf)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	for {
+		select {
+		case pong := <-d.ch:
+			if pong.UUID == ping.UUID {
+				return nil
+			}
+		case <-ctx.Done():
+			return trace.Wrap(ctx.Err())
+		}
+	}
+}
+
 // proxyWebsocketConn does a bidrectional copy between the websocket
 // connection to the browser (ws) and the mTLS connection to Windows
 // Desktop Serivce (wds)
-func proxyWebsocketConn(ws *websocket.Conn, wds net.Conn) error {
+func proxyWebsocketConn(ws *websocket.Conn, wds net.Conn, log *slog.Logger, version string) error {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Should we pass in the context from `createDekstopConnection` rather than using a background context here?

Reply:
We should, I've added it